src/main/java/pl/damianszczepanik/jenkins/buildhistorymanager/model/actions/DeleteArtifactsMatchingPatternsAction.java

```diff
@@ -0,0 +1,147 @@
+package pl.damianszczepanik.jenkins.buildhistorymanager.model.actions;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.DirectoryStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import org.jenkinsci.remoting.RoleChecker;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
+
+import hudson.FilePath;
+import hudson.FilePath.FileCallable;
+import hudson.Util;
+import hudson.model.Run;
+import hudson.remoting.VirtualChannel;
+import jenkins.util.VirtualFile;
+
+/**
+ * Deletes the artifacts matching patterns.
+ */
+public class DeleteArtifactsMatchingPatternsAction extends Action {
+    private static final Logger LOG = Logger.getLogger(DeleteArtifactsMatchingPatternsAction.class.getName());
+
+    private String includePatterns;
+    private String excludePatterns;
+
+    @DataBoundConstructor
+    public DeleteArtifactsMatchingPatternsAction() {
+        // Jenkins stapler requires to have public constructor with @DataBoundConstructor
+    }
+
+    public String getIncludePatterns() {
+        return includePatterns;
+    }
+
+    @DataBoundSetter
+    public void setIncludePatterns(String includePatterns) {
+        this.includePatterns = includePatterns;
+    }
+
+    public String getExcludePatterns() {
+        return excludePatterns;
+    }
+
+    @DataBoundSetter
+    public void setExcludePatterns(String excludePatterns) {
+        this.excludePatterns = excludePatterns;
+    }
+
+    static boolean isDirectoryEmpty(Path path) throws IOException {
+        try (DirectoryStream<Path> directoryStream = Files.newDirectoryStream(path)) {
+            return !directoryStream.iterator().hasNext();
+        }
+    }
+
+    // if 'file' is on a different node, this FileCallable will be transferred to that node and executed there.
+    public static final class DeleteFileCallable implements FileCallable<Void> {
+        private static final long serialVersionUID = -7712623306942117400L;
```

Comment:
if this class is static I believe you don't store it and if so then serialVersionUID is probably not needed in that case

Reply:
At first, I removed the `serialVersionUID` from the code, but I got warnings, as you can see in the screenshot. After studying, I found that Java makes this ID automatically if we don't provide it. Since our code uses `FileCallable`, which is connected to `Serializable`, it's like a rule that classes using Serializable should have a  `serialVersionUID` field . So, I had to add it to address the warnings. 
![image](https://github.com/jenkinsci/build-history-manager-plugin/assets/90399606/33e3c2b6-ea1d-4401-b3ad-4983f669dbdd)